src/stories/Tooltip.stories.tsx

```diff
@@ -22,10 +23,179 @@ export default {
   ],
 } as Meta
 
-export const IconButtonTooltip = () => (
+export const CurrentTooltip = () => (
   <Box sx={{p: 5}}>
-    <Tooltip aria-label="Search">
+    <Tooltip text="Search">
       <IconButton icon={SearchIcon} aria-label="Search" />
     </Tooltip>
   </Box>
 )
+
+// As a label for an IconButton
+export const Tooltip2LabelTypeTooltip = () => (
+  <Box sx={{p: 5}}>
+    <Tooltip2>
+      <IconButton icon={SearchIcon} aria-label="Search" />
```

Comment:
It'd be great if we could move the `aria-label` up a level and not support inferring it from the child. Is this something that currently exists in the component that we're needing to support still?

Reply:
I responded [here](https://github.com/primer/react/pull/3032#discussion_r1142806481) 🙂 Thanks again for pointing this out! 

Question - For this type of usage

```
<Tooltip2 aria-label="Search tooltip">
    <IconButton icon={SearchIcon} aria-label="Search button itself" />
</Tooltip2>
```

With the current remediations, the button will end up having both `aria-label` and `aria-labelledby` attribute and I have [this line of code](https://github.com/primer/react/pull/3032/files#diff-3516f4ecd704760b75cc577257c4c3e36f6830d590bdf4c37364d4edde78547dR234) to remove the `aria-label` from the trigger child as it will have `aria-labelledby` but I don't know this is a good practise? 

I tested it on VoiceOver and it seems like when both `aria-label` and `aria-labelledby` are present, it will disregard the `aria-label`. So I am a bit not sure what would be the best practise here 🤔

Reply:
@broccolinisoup the `IconButton` is definitely a tough situation. Ideally, I think `IconButton` should render a `Tooltip` over the underlying `Button` so that this behavior is built-in by default. In other words, one would write:

```tsx
<IconButton icon={SearchIcon} aria-label="Search button itself" />
```

And this would include the tooltip by default.

I personally worry about `child.props` since someone could add a wrapper component in between and still retain the incorrect usage 😞 For example,

```jsx
<Tooltip2 aria-label="Search tooltip">
  <CustomIconButton />
</Tooltip2>

function CustomIconButton() {
  return <IconButton icon={SearchIcon} aria-label="Search button itself" />
}
```

I think if we can require the `child` to accept a `ref`, then we should be able to warn for either usage (basically is any child labelled) using that `ref`. e.g.

```tsx
function Tooltip({ children }) {
  const child = React.Children.only(children);
  const ref = React.useRef(null);

  // ...

  if (__DEV__) {
    useEffect(() => {
      // Check contents of `ref` for label usage
    });
  }

  // ...
  {React.cloneElement(child, { ref, /* ... */ })}
  // ...
}
```

Reply:
🔥 I love this so much. This is a great point!